docs/blueprint/EP039.rst

```diff
@@ -0,0 +1,280 @@
+  - Initialization
+    - The kafka_napp starts execution through the setup() function.
+    - Within setup(), an instance of KafkaSendOperations is created.
+    - If the specified Kafka topic does not exist, KafkaSendOperations ensures its creation.
+  - Event Listening and Filtering
+    - The kafka_napp listens to all Kytos network events by subscribing to the event bus.
+    - It checks each event against a predefined configuration to determine whether it should be processed.
+    - Events that are not configured for processing are filtered out and discarded.
+  - Key Components
+    - Event Listener: Listens to all Kytos events and applies filtering rules.
+    - AsyncScheduler: Adds tasks to an asynchronous loop for further processing.
+
+
+III. How kafka_napp interacts with Kafka
+===============================================
+
+1. The kafka_napp also serves as an asynchronous producer of events, producing serialization and network submission functionality.
+
+- Publishing to Kafka
+  - Serialization
+    - Once filtered, kafka_napp will add a task to be completed in the asynchronous event loop.
+    - Events are serialized and published to the corresponding Kafka topic.
+      - The topic name is configured in `settings.py`.
+      - The defaut is `event_logs`.
+      - Kafka ensures these events are durably stored and available for consumption by downstream services.
+  - Production
+    - Events are sent in batches asynchronously, ensuring maximum efficiency in network requests and processing.
+  - Key Components
+    - KafkaSendOperations: Handles the creation of Kafka topics (if necessary) and manages event publishing.
+
+
+IV. Configurations in settings.py
+==============================
+
+IV. Configurations in settings.py
+
+1. This section describes the key configuration parameters used by the kafka_napp application. These constants define how the application connects to Kafka, manages topics, handles message delivery, and filters events.
+
+- Key Configuration Constants
+  - BOOTSTRAP_SERVERS
+    - Description: Specifies the Kafka server's address (hostname and port).
+    - Usage: This setting is critical for establishing the connection with the Kafka broker.
+  - ACKS
+    - Description: Determines the level of message acknowledgements required by Kafka.
+    - Available Options:
+      - 0: No acknowledgements required.
+      - 1: Only the leader broker must acknowledge.
+      - "all": All in-sync replicas must acknowledge the message.
+      - Usage: Controls the reliability and durability of message delivery.
+      - Note: This may increase latency between the application during sends. To add this functionality, ensure tolerance for when the application is sending to kafka but the also wants to enqueue new requests.
+  - DEFAULT_NUM_PARTITIONS
+    - Description: Sets the default number of partitions per Kafka topic.
+    - Usage: Influences the parallelism and throughput of message processing.
+  - IGNORED_EVENTS
```

Comment:
This is a good starting point for this iteration. We should consider supporting a few non expensive regexes, say for instance if you want to filter all events from topology, currently it'd be a pain in the neck to do it. With a regex, you could do `kytos/topology.*`. Something to think as a minor feature in the future.